src/main/java/ru/skypro/homework/controller/AdsController.java

```diff
@@ -0,0 +1,106 @@
+package ru.skypro.homework.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
+import ru.skypro.homework.dto.AdDto;
+import ru.skypro.homework.dto.AdsDto;
+import ru.skypro.homework.dto.CreateOrUpdateAdDto;
+import ru.skypro.homework.dto.ExtendedAdDto;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+@CrossOrigin(value = "http://localhost:3000")
+@RestController
+@RequestMapping("/ads")
+@Tag(name = "Объявления")
+public class AdsController {
+    @DeleteMapping("/{id}")
+    @Operation(summary = "Удаление объявления",
+            description = "Удаление объявления по идентификационному номеру авторизованным пользователем")
+    @ApiResponse(responseCode = "204", description = "No Content")
+    @ApiResponse(responseCode = "401", description = "Unauthorized")
+    @ApiResponse(responseCode = "403", description = "Forbidden")
+    @ApiResponse(responseCode = "404", description = "Not Found")
+    public ResponseEntity<Void> removeAd(@PathVariable("id") Integer id) {
+        // Реализация удаления объявления
+        return ResponseEntity.noContent().build();
+    }
+
+    @GetMapping
+    @Operation(summary = "Получение всех объявлений",
+            description = "Получение количества и списка всех объявлений")
+    @ApiResponse(responseCode = "200", description = "OK")
+    public ResponseEntity<AdsDto> getAllAds() {
+        AdsDto allAdsDtoList = new AdsDto(0, new ArrayList<>());
+        return ResponseEntity.ok(allAdsDtoList);
+    }
+
+    @PostMapping(consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
+    @Operation(summary = "Добавление объявления",
+            description = "Добавление изображения и всех полей объявления")
+    @ApiResponse(responseCode = "201", description = "Created")
+    @ApiResponse(responseCode = "401", description = "Unauthorized")
+    public ResponseEntity<AdDto> addAd(@RequestBody CreateOrUpdateAdDto createOrUpdateAdDto,
+                                       @RequestParam MultipartFile image) throws IOException {
```

Comment:
```suggestion
    public ResponseEntity<AdDto> addAd(@RequestPart("properties") CreateOrUpdateAdDto createOrUpdateAdDto,
                                       @RequestPart MultipartFile image) throws IOException {
```
неправильные аннотации и также названия частей запроса

Reply:
исправлено